diff --git a/Source/ThirdParty/ANGLE/src/common/bitset_utils.h b/Source/ThirdParty/ANGLE/src/common/bitset_utils.h
index e9592d7f7422..2f8c007fdf19 100644
--- a/Source/ThirdParty/ANGLE/src/common/bitset_utils.h
+++ b/Source/ThirdParty/ANGLE/src/common/bitset_utils.h
@@ -687,6 +687,11 @@ class BitSetArray final
     // Produces a mask of ones up to the "x"th bit.
     constexpr static BitSetArray Mask(std::size_t x);
 
+    template <std::size_t BitSetSize, std::enable_if_t<(BitSetSize < 64), int> = 0>
+    constexpr void initFromValue(uint64_t value);
+    template <std::size_t BitSetSize, std::enable_if_t<(BitSetSize >= 64), int> = 0>
+    constexpr void initFromValue(uint64_t value);
+
   private:
     static constexpr std::size_t kDefaultBitSetSizeMinusOne = priv::kDefaultBitSetSize - 1;
     static constexpr std::size_t kShiftForDivision =
@@ -713,26 +718,8 @@ constexpr BitSetArray<N>::BitSetArray()
 template <std::size_t N>
 constexpr BitSetArray<N>::BitSetArray(uint64_t value)
 {
     reset();
-
-    if (priv::kDefaultBitSetSize < 64)
-    {
-        size_t i = 0;
-        for (; i < kArraySize - 1; ++i)
-        {
-            value_type elemValue =
-                value & priv::BaseBitSetType::Mask(priv::kDefaultBitSetSize).bits();
-            mBaseBitSetArray[i] = priv::BaseBitSetType(elemValue);
-            value >>= priv::kDefaultBitSetSize;
-        }
-        value_type elemValue = value & kLastElementMask;
-        mBaseBitSetArray[i]  = priv::BaseBitSetType(elemValue);
-    }
-    else
-    {
-        value_type elemValue = value & priv::BaseBitSetType::Mask(priv::kDefaultBitSetSize).bits();
-        mBaseBitSetArray[0]  = priv::BaseBitSetType(elemValue);
-    }
+    initFromValue<priv::kDefaultBitSetSize>(value);
 }
 
 template <std::size_t N>
@@ -1089,6 +1076,32 @@ constexpr BitSetArray<N> BitSetArray<N>::Mask(std::size_t x)
 
     return result;
 }
+
+template <std::size_t N>
+template <std::size_t BitSetSize, std::enable_if_t<(BitSetSize < 64), int>>
+constexpr void BitSetArray<N>::initFromValue(uint64_t value)
+{
+    static_assert(BitSetSize == 32, "Expected 32 bit size");
+    size_t i = 0;
+    for (; i < kArraySize - 1; ++i)
+    {
+        value_type elemValue = value & priv::BaseBitSetType::Mask(BitSetSize).bits();
+        mBaseBitSetArray[i]  = priv::BaseBitSetType(elemValue);
+        value >>= BitSetSize;
+    }
+    value_type elemValue = value & kLastElementMask;
+    mBaseBitSetArray[i]  = priv::BaseBitSetType(elemValue);
+}
+
+template <std::size_t N>
+template <std::size_t BitSetSize, std::enable_if_t<(BitSetSize >= 64), int>>
+constexpr void BitSetArray<N>::initFromValue(uint64_t value)
+{
+    static_assert(BitSetSize == 64, "Expected 64 bit size");
+    value_type elemValue = value & priv::BaseBitSetType::Mask(BitSetSize).bits();
+    mBaseBitSetArray[0]  = priv::BaseBitSetType(elemValue);
+}
+
 }  // namespace angle
 
 template <size_t N, typename BitsT, typename ParamT>
